src/client/Microsoft.Identity.Client.Brokers/RuntimeBroker.cs

```diff
@@ -0,0 +1,392 @@
+            AuthenticationRequestParameters authenticationRequestParameters,
+            AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
+        {
+            MsalTokenResponse msalTokenResponse = null;
+
+            //need to provide a handle
+            if (_parentHandle == IntPtr.Zero)
+            {
+                throw new MsalClientException(
+                    "window_handle_required",
+                    "Public Client applications wanting to use WAM need to provide their window handle. Console applications can use GetConsoleWindow Windows API for this.");
+            }
+
+            //if OperatingSystemAccount is passed then we use the user signed-in on the machine
+            if (PublicClientApplication.IsOperatingSystemAccount(authenticationRequestParameters.Account))
+            {
+                return await AcquireTokenInteractiveDefaultUserAsync(authenticationRequestParameters, acquireTokenInteractiveParameters).ConfigureAwait(false);
+            }
+
+            var cancellationToken = authenticationRequestParameters.RequestContext.UserCancellationToken;
+                
+            _logger.Verbose("[WamBroker] Using Windows account picker.");
+
+            using (var core = new NativeInterop.Core())
+            using (var authParams = WamAdapters.GetCommonAuthParameters(authenticationRequestParameters, _wamOptions.MsaPassthrough))
+            {
+                //Login Hint
+                string loginHint = authenticationRequestParameters.LoginHint ?? authenticationRequestParameters?.Account?.Username;
+
+                if (!string.IsNullOrEmpty(loginHint))
+                {
+                    _logger.Verbose("[WamBroker] AcquireTokenInteractive - account information provided. Trying to find a Windows account that matches.");
+                }
+                else
+                {
+                    _logger.Verbose("[WamBroker] Account information was not provided. Using an account picker.");
+                }
+
+                using (var result = await core.SignInInteractivelyAsync(
+                    _parentHandle,
+                    authParams,
+                    authenticationRequestParameters.CorrelationId.ToString("D"),
+                    loginHint,
+                    cancellationToken).ConfigureAwait(false))
+                {
+                    if (result.IsSuccess)
+                    {
+                        msalTokenResponse = WamAdapters.ParseRuntimeResponse(result, authenticationRequestParameters, _logger);
+                        _logger.Verbose("[WamBroker] Successfully retrieved token.");
+
+                    }
+                    else
+                    {
+                        _logger.Error($"[WamBroker] Could not login interactively. {result.Error}");
+                        WamAdapters.CreateWamErrorResponse(result, authenticationRequestParameters, _logger);
```

Comment:
Rename this to `ThrowExceptionFromWamError` or smth that suggests that this method will throw.